common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/RestClientVrops.java

```diff
@@ -389,26 +469,36 @@ public List<PolicyDTO.Policy> getAllPolicies() {
 		HttpHeaders headers = new HttpHeaders();
 		headers.setContentType(MediaType.APPLICATION_JSON_UTF8);
 
-		// Required header for internal API
-		headers.set(INTERNAL_API_HEADER_NAME, Boolean.TRUE.toString());
+		// older vROPs versions use internal API for policies, thus internal header needs to be set.
+		if (!this.isVersionAbove812()) {
+			headers.set(INTERNAL_API_HEADER_NAME, Boolean.TRUE.toString());			
+		}		
 		HttpEntity<String> entity = new HttpEntity<>(headers);
 		ResponseEntity<String> response;
 		try {
-			UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromUri(new URI(getURIBuilder().setPath(POLICIES_API).toString()));
+			UriComponentsBuilder uriBuilder;
+			// for newer vROPs versions the policies API is no longer internal.
+			if (this.isVersionAbove812()) {
+				uriBuilder = UriComponentsBuilder.fromUri(getURI(getURIBuilder().setPath(POLICIES_FETCH_PUBLIC_API)));
+			} else {
+				uriBuilder = UriComponentsBuilder.fromUri(getURI(getURIBuilder().setPath(POLICIES_FETCH_INTERNAL_API)));
+				
+			}			
 			uriBuilder.queryParam("pageSize", DEFAULT_PAGE_SIZE);
 			URI restUri = uriBuilder.build().toUri();
 			response = restTemplate.exchange(restUri, HttpMethod.GET, entity, String.class);
 		} catch (HttpClientErrorException e) {
 			if (HttpStatus.NOT_FOUND.equals(e.getStatusCode())) {
 				return new ArrayList<>();
 			}
-			throw new RuntimeException(String.format("Error ocurred trying to fetching policies. Message: %s", e.getMessage()));
-		} catch (URISyntaxException e) {
-			throw new RuntimeException(String.format("Error building REST URI to fetch policies. Message: %s", e.getMessage()));
+			throw new RuntimeException(String.format("HTTP error ocurred trying to fetching policies. Message: %s", e.getMessage()));
+		} catch (Exception e) {
+			throw new RuntimeException(String.format("General error building REST URI to fetch policies. Message: %s", e.getMessage()));
```

Comment:
Make sure we did not swallowed original error - `e.getMessage()` will not include inner exception or stack trace and etc.
Always provide original exception as inner one when throw new Exception().